pallets/erc20-xcm-bridge/src/erc20_matcher.rs

```diff
@@ -67,6 +67,28 @@ impl<Erc20MultilocationPrefix: Get<MultiLocation>> Erc20Matcher<Erc20Multilocati
 			_ => Err(()),
 		}
 	}
+	pub(crate) fn matches_gas_limit(multiasset: &MultiAsset) -> Option<u64> {
+		let multilocation = match (&multiasset.fun, &multiasset.id) {
+			(Fungible(_), Concrete(ref id)) => id,
+			_ => return None,
+		};
+
+		// let prefix = Erc20MultilocationPrefix::get();
+		match multilocation.interior().into_iter().next_back() {
+			Some(Junction::GeneralKey { length, data }) => {
+				let content = core::str::from_utf8(data).expect("f");
```

Comment:
The onchain runtime should never panic, you can't use expect here, you should return None instead